docker-app/qfieldcloud/core/models.py

```diff
@@ -1337,22 +1337,25 @@ def latest_package_job_for_user(self, user: User) -> PackageJob | None:
 
         secret_qs = Secret.objects.filter(secret_filters)
 
-        jobs_qs = (
-            self.jobs.annotate(has_user_secret=Exists(secret_qs))
-            .filter(Q(has_user_secret=False) | Q(triggered_by=user))
-            .order_by("-created_at")
+        jobs_qs = self.jobs.annotate(has_user_secret=Exists(secret_qs)).filter(
+            Q(has_user_secret=False) | Q(triggered_by=user)
         )
 
-        jobs_qs = jobs_qs.order_by("-created_at")
-
-        return (
-            jobs_qs.filter(
-                type=Job.Type.PACKAGE,
-            )
-            .order_by("-created_at")
-            .first()
+        return jobs_qs.filter(
+            type=Job.Type.PACKAGE,
         )
 
+    def latest_package_job_for_user(self, user: User) -> PackageJob | None:
```

Comment:
Why don't we add latest finished package  job as a function as you already use it twice? Also do you use the current function somewhere?

Reply:
> So I presume that if the latest package job is ongoing and not finished yet, needs_repackaging will be true. Is this what we want ?

I think at that stage in `needs_repackaging` we have no idea if the status of the job will be success, so it is always safer to trigger more jobs rather than giving false promises.